File: web/src/utils/WireUrlResolver.test.ts
import { describe, expect, it } from 'vitest';
import { resolveWireAssetUrl } from './WireUrlResolver';

describe('resolveWireAssetUrl', () => {
  it('leaves absolute URLs unchanged', () => {
    expect(resolveWireAssetUrl('https://cdn.example.com/a.png', 'http://localhost:8080')).toBe(
      'https://cdn.example.com/a.png',
    );
  });

  it('joins relative paths to base', () => {
    expect(resolveWireAssetUrl('/sdui-demo/team.svg?v=1', 'http://10.0.2.2:8080')).toBe(
      'http://10.0.2.2:8080/sdui-demo/team.svg?v=1',
    );
  });

  it('returns relative path when base is empty', () => {
    expect(resolveWireAssetUrl('/sdui-demo/x.svg', '')).toBe('/sdui-demo/x.svg');
  });
});

File: android/sdui-core/src/test/java/com/nba/sdui/core/network/WireUrlResolverTest.kt
package com.nba.sdui.core.network

import org.junit.Assert.assertEquals
import org.junit.Test

class WireUrlResolverTest {
    @Test
    fun `leaves absolute urls unchanged`() {
        assertEquals(
            "https://cdn.example.com/a.png",
            WireUrlResolver.resolve("https://cdn.example.com/a.png", "http://10.0.2.2:8080")
        )
    }

    @Test
    fun `prefixes root relative demo assets`() {
        assertEquals(
            "http://10.0.2.2:8080/sdui-demo/card-wide.svg?v=heroarena",
            WireUrlResolver.resolve("/sdui-demo/card-wide.svg?v=heroarena", "http://10.0.2.2:8080")
        )
    }
}
